pkg/cli/admin/rebootmachineconfigpool/reboot_machine_config_pool.go

```diff
@@ -0,0 +1,200 @@
+			DryRun:       []string{metav1.DryRunAll},
+			FieldManager: RebootMachineConfigPoolFieldManager,
+		}
+	}
+	return metav1.CreateOptions{
+		FieldManager: RebootMachineConfigPoolFieldManager,
+	}
+}
+
+func (r *RebootMachineConfigPoolRuntime) updateOptions() metav1.UpdateOptions {
+	if r.dryRun {
+		return metav1.UpdateOptions{
+			DryRun:       []string{metav1.DryRunAll},
+			FieldManager: RebootMachineConfigPoolFieldManager,
+		}
+	}
+	return metav1.UpdateOptions{
+		FieldManager: RebootMachineConfigPoolFieldManager,
+	}
+}
+
+func (r *RebootMachineConfigPoolRuntime) rebootMachineConfigPool(ctx context.Context, machineConfigPool *MachineConfigPool) error {
+	// do something better.
+	if machineConfigPool.Spec.MachineConfigSelector == nil {
+		return fmt.Errorf("machineconfigpools/%v cannot be rebooted with this command due to missing .spec.machineConfigSelector", machineConfigPool.Name)
+	}
+	if len(machineConfigPool.Spec.MachineConfigSelector.MatchExpressions) > 0 {
+		// TODO probably add an option to specify labels somehow?  Maybe its not common in the field?
+		return fmt.Errorf("machineconfigpools/%v cannot be rebooted with this command due to usage of .spec.machineConfigSelector.matchExpressions", machineConfigPool.Name)
+	}
+	if len(machineConfigPool.Spec.MachineConfigSelector.MatchLabels) == 0 {
+		return fmt.Errorf("machineconfigpools/%v cannot be rebooted with this command due to missing .spec.machineConfigSelector.matchLabels", machineConfigPool.Name)
+	}
+
+	machineConfig := restartTemplate.DeepCopy()
+	if machineConfig.Labels == nil {
+		machineConfig.Labels = map[string]string{}
+	}
+	machineConfig.Name = fmt.Sprintf("95-certrotation-rebooter-%s", machineConfigPool.Name)
+	for k, v := range machineConfigPool.Spec.MachineConfigSelector.MatchLabels {
+		machineConfig.Labels[k] = v
+	}
+
+	finalObject := &MachineConfig{}
+	currMachineConfigUnstructured, err := r.DynamicClient.Resource(machineConfigResource).Get(ctx, machineConfig.Name, metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+		machineConfigUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(machineConfig)
+		if err != nil {
+			return fmt.Errorf("not a secret: %w", err)
+		}
+		toCreate := &unstructured.Unstructured{Object: machineConfigUnstructured}
+
+		finalUnstructured, err := r.DynamicClient.Resource(machineConfigResource).Create(ctx, toCreate, r.createOptions())
+		if err != nil {
+			return err
+		}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(finalUnstructured.Object, finalObject); err != nil {
+			return err
+		}
+
+	case err != nil:
+		return err
+
+	default:
+		currMachineConfig := &MachineConfig{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(currMachineConfigUnstructured.Object, currMachineConfig); err != nil {
+			return err
+		}
+		machineConfig.ResourceVersion = currMachineConfig.ResourceVersion
+		// TODO figure out how to read the existing int and set a bigger one.
```

Comment:
this TODO is quite important

Reply:
I think probably having an annotation on the MC object to match the contents would be easiest, and better to track instead of reading the machineconfig directly. Those can be updated in tandem. WDYT?

Reply:
> I think probably having an annotation on the MC object to match the contents would be easiest, and better to track instead of reading the machineconfig directly. Those can be updated in tandem. WDYT?

no opinion.  I don't know how to change the int either way.